spec/horoshop/authorization_spec.rb

```diff
@@ -3,54 +3,53 @@
 
 describe Horoshop::Authorization do
-  let(:instance) { double('Instance', username: 'user', password: 'pass', url: 'http://api.horosop.com') }
+  subject(:auth) { described_class.new(instance) }
+
+  let(:instance) { instance_double('Instance', username: 'user123', password: 'pass123', url: 'http://api.horosop.com') }
 
   before do
     allow(instance).to receive(:token=)
     allow(instance).to receive(:expiration_timestamp=)
-    allow(Time).to receive(:now).and_return(Time.now)
+    allow(Time).to receive(:now).and_return(Time.parse('2024-02-21 12:42:59 +0200'))
   end
 
-  subject(:authorization) { described_class.new(instance) }
-
   describe '#authorize' do
     context 'when authorization is successful' do
       let(:successful_response) do
         {
-          body: { 'status' => 'OK', 'response' => { 'token' => 'valid_token' } }.to_json,
+          body: { 'status' => 'OK', 'response' => { 'token' => '123qwe4' } }.to_json,
           headers: { 'Content-Type' => 'application/json' }
         }
       end
 
       before do
         stub_request(:post, 'http://api.horosop.com/api/auth/')
-          .with(body: { login: 'user', password: 'pass' })
+          .with(body: { login: 'user123', password: 'pass123' })
           .to_return(successful_response)
       end
 
       it 'sets the token and expiration_timestamp on the instance' do
-        authorization.authorize
-        expect(instance).to have_received(:token=).with('valid_token')
-        expect(instance).to have_received(:expiration_timestamp=).with(an_instance_of(Time))
+        expect(instance).to receive(:token=).with('123qwe4')
+        expect(instance).to receive(:expiration_timestamp=).with(Time.now + 600)
+        auth.authorize
       end
     end
 
     context 'when authorization fails' do
       before do
         stub_request(:post, 'http://api.horosop.com/api/auth/')
-          .with(body: { login: 'user', password: 'pass' })
+          .with(body: { login: 'user123', password: 'pass123' })
           .to_return(
             status: 200,
-            body: { 'status' => 'ERROR', 'response' => { 'message' => 'some_message' } }.to_json
+            body: { 'status' => 'ERROR', 'response' => { 'message' => 'error message' } }.to_json
           )
       end
 
       it 'does not set the token and expiration_timestamp on the instance' do
-        response = authorization.authorize
-        expect(JSON.parse(response)).to include('status' => 'ERROR')
-        expect(JSON.parse(response)).to include('response' => { 'message' => 'some_message' })
-        expect(instance).not_to have_received(:token=)
-        expect(instance).not_to have_received(:expiration_timestamp=)
+        expect(instance).not_to receive(:token=)
+        expect(instance).not_to receive(:expiration_timestamp=)
+        expect(JSON.parse(auth.authorize)).to include('status')
+        expect(JSON.parse(auth.authorize)).to include('response' => match('message' => 'error message'))
```

Comment:
1. lets add expect that returned method is object
2. Below lines always will be true. You do not call anytime authorize method. You should just add ` after {auth.authorize}` and wrap every expect into separated block `it`
`
expect(instance).not_to receive(:token=)
expect(instance).not_to receive(:expiration_timestamp=)
 `